### PR TITLE
In-app reviews

### LINKED ISSUE #12605: Add an in-app review prompt
Google recently released a [new In-App Review API for Google Play reviews](https://android-developers.googleblog.com/2020/08/in-app-review-api.html). From their [developer docs](https://developer.android.com/guide/playcore/in-app-review):

> The Google Play In-App Review API lets you prompt users to submit Play Store ratings and reviews without the inconvenience of leaving your app or game.
>
> Generally, the in-app review flow can be triggered at any time throughout the user journey of your app. During the flow, the user has the ability to rate your app using the 1 to 5 star system and to add an optional comment. Once submitted, the review is sent to the Play Store and eventually displayed.

Naturally, there are some restrictions to consider:

> To protect user privacy and avoid API misuse, there are strict guidelines that your app should follow about when to request in-app reviews and the design of the review prompt.

Let's figure out how to integrate this into WPAndroid!

/cc @mattmiklic @frosty @yaelirub

### PR DESCRIPTION
**Pull request by irfano**
Fixes #12605 

This adds [in-app reviews feature](https://developer.android.com/guide/playcore/in-app-review). The prompt will display after the second successful publishing (post or story). It must be a new post, and editing doesn't count.  

https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/4914894f-3819-4a89-96ed-a7af502611fe

-----

## To Test:

Testing this feature is not available within the PR. It will be tested outside of the PR.
Internal link: p1702317430036659-slack-C441E3YTS

1. Install the app from the link.
2. Log in.
3. Publish a post or story. 
4. Verify that the blogging reminder is displayed.
5. Publish another post or story.
6. Verify that the in-app previews prompt is displayed. (Submitting the review may not be available due to the testing environment.)

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

7. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Nothing

8. What automated tests I added (or what prevented me from doing so)

    - `ReviewViewModelTest`

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by ravishanker — APPROVED** [review]
Awesome work 🚀   Submit is disabled due to test env.  We may need to monitor this to find an optimum event when this can be triggered.  For now though, on second post seems reasonable to start with.

File: docs/test_instructions_per_dependency_update.md [hunk 1]
```diff
@@ -421,6 +422,19 @@ Step.3:
 
 -----
 
+### PlayReview [[googlePlayReviewVersion](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/build.gradle)] <a name="playreview"></a>
+
+<details>
+    <summary>1. In app reviews</summary>
+
+- Perform a clean install.
+- Publish three (`ReviewViewModel.TARGET_COUNT_POST_PUBLISHED`) new posts or stories.
```
Comment (ravishanker) [inline docs/test_instructions_per_dependency_update.md]: Is this supposed to be `Publish two`?

**Comment by irfano** [comment]
> We may need to monitor this to find an optimum event when this can be triggered.

Monitoring won't be easy because checking the Play Store rating is the only method to do it. I'll post a P2 regarding this.

File: docs/test_instructions_per_dependency_update.md (hunk 1 above)
Comment (irfano) [reply docs/test_instructions_per_dependency_update.md]: I intentionally typed three to confirm that it does not crash after `TARGET_COUNT_POST_PUBLISHED` is reached.  I updated the parentheses information from `ReviewViewModel.TARGET_COUNT_POST_PUBLISHED` to `ReviewViewModel.TARGET_COUNT_POST_PUBLISHED + 1`: 240ed435cf1c264a073f954a02706a1c085dfc0f

**Pull request merged**